processing/src/main/java/io/druid/guice/ExtensionsConfig.java

```diff
@@ -33,10 +34,6 @@
   @NotNull
   private boolean searchCurrentClassloader = true;
 
-  @JsonProperty
-  @NotNull
-  private List<String> coordinates = ImmutableList.of();
```

Comment:
If someone is updating their cluster to the latest binaries, it could be easy for them to miss the release notes that the `druid.extensions.coordinates` changed. And it can cause late failures.

Since the config change is not handled at start time, and there is no migration method to warn people of the config change (other than release notes) AND failure to migrate to the new config can cause node failure due to extensions not being present when they are expected to be present, that would be an incompatible change.

Personally, I'm ok with eliminating the setting provided that there is an informative error on startup (or even better, an informative warning and transparent handling of migration until next major version release). Would be worth getting input from the other contributors though.